utils/common/src/main/java/brooklyn/util/javalang/AggregateClassLoader.java

```diff
@@ -69,8 +70,25 @@ public void add(int index, ClassLoader classLoader) {
         if (classLoader != null) classLoaders.add(index, classLoader);
     }
     
-    /** Returns the _live_ (and modifiable) list of classloaders 
-     * @return */ 
+    /** Resets the classloader shown here to be the given set */
+    public void reset(Collection<? extends ClassLoader> newClassLoaders) {
+        synchronized (classLoaders) {
```

Comment:
Clever adding all to end, and then removing `count` from start. However, it will break if there is a concurrent call to `add(int index, ClassLoader classLoader)`.

Personally I'd prefer a simpler synchronization model, that does not try to be clever.